tests/trainer/test_data_loading.py

```diff
@@ -335,3 +335,19 @@ def test_pre_made_batches():
     loader = DataLoader(RandomDataset(32, 10), batch_size=None)
     trainer = Trainer(fast_dev_run=1)
     trainer.predict(LoaderTestModel(), loader)
+
+
+@RunIf(skip_windows=True)
+def test_distributed_sampler_with_overfit_batches(tmpdir):
+    model = BoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        overfit_batches=1,
+        fast_dev_run=1,
+        strategy="ddp_find_unused_parameters_false",
```

Comment:
a particular reason for this choice? "ddp_spawn" would normally suffice

Reply:
But `num_processes` is 1?

Reply:
This was the old mechanism to force "DDP" on "CPU"

nowadays just `strategy="ddp"` works on CPU. I guess this was copied from an old test

Reply:
yeah.. looks like the case.. we don't update the provided strategy in case num_processes stays 1

Reply:
it's actually a bug here I think. when you pass just ddp or ddp_spawn, it defaults to single device plugin but when provided with registry arguments it doesn't.

Reply:
I don't think so.. with `ddp` it goes on single device plugin but with `ddp_something_false` it stays in ddp